src/app/services/actions/autocomplete-action-creators.ts

```diff
@@ -0,0 +1,91 @@
+import { IAction } from '../../../types/action';
+import { IRequestOptions } from '../../../types/request';
+import { AUTOCOMPLETE_FETCH_ERROR, AUTOCOMPLETE_FETCH_PENDING, AUTOCOMPLETE_FETCH_SUCCESS } from '../redux-constants';
+
+export function fetchAutocompleteSuccess(response: object): IAction {
+  return {
+    type: AUTOCOMPLETE_FETCH_SUCCESS,
+    response,
+  };
+}
+
+export function fetchAutocompleteError(response: object): IAction {
+  return {
+    type: AUTOCOMPLETE_FETCH_ERROR,
+    response,
+  };
+}
+
+export function fetchAutocompletePending(): any {
+  return {
+    type: AUTOCOMPLETE_FETCH_PENDING
+  };
+}
+
+export function fetchAutoCompleteOptions(url: string): Function {
+  return async (dispatch: Function, getState: Function) => {
+    const devxApi = getState().devxApi;
+    const sampleQuery = getState().sampleQuery;
+    const headers = {
+      'Content-Type': 'application/json',
+    };
+
+    let permissionsUrl = `${devxApi}/openapi`;
+    permissionsUrl = `${permissionsUrl}?url=/${url}&style=geautocomplete&format=json`;
+
+    const options: IRequestOptions = { headers };
+
+    dispatch(fetchAutocompletePending());
+
+    try {
+      const response = await fetch(permissionsUrl, options);
+      if (response.ok) {
+        const autoCompleteOptions = await response.json();
+        const params = {
+          options: autoCompleteOptions,
+          url,
+          verb: sampleQuery.selectedVerb.toLowerCase()
+        };
+        const reduced = getReducedVersion(params);
+
+        return dispatch(fetchAutocompleteSuccess(reduced));
+      }
+
+      throw (response);
+
+    } catch (error) {
+      return dispatch(fetchAutocompleteError(error));
```

Comment:
Instead of throwing at line 53 then catching a few lines later, we can just dispatch if `!response.ok`

```js
if (!response.ok) {
  return dispatch(...)
}

...
return dispatch(fetchAutocompleteSuccess)
```

Reply:
Covered in this [commit](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/pull/489/commits/48ee2872a62de115af7907d7d7c9253c360b4651)